### PR TITLE
Remove unused Dockerfile.rhel7 file

### PR DESCRIPTION
**Pull request by martinkennelly**
This can be accomplished because of PR:
https://github.com/openshift/release/pull/38079

cc @jluhrsen

### CODE REVIEW

File: Dockerfile.rhel7 [hunk 1]
```diff
@@ -1,14 +0,0 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.19-openshift-4.14 AS builder
```
Comment (martinkennelly) [inline Dockerfile.rhel7:1]: @jluhrsen There is a different here between Dockerfile and this - in Dockerfile its: 
```
FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.14 AS builder
```

**Comment by dcbw** [comment]
Ugh, we actually *do* use Dockerfile.rhel7:

[ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml](https://github.com/openshift/release/blob/master/ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml#L30)

So we have to first change that to `Dockerfile` and then we can kill the .rhel7 version.

File: Dockerfile.rhel7 (hunk 1 above)
Comment (jluhrsen) [reply Dockerfile.rhel7:1]: so I think we need this to fix it back up:
https://github.com/openshift/release/pull/39281

**Comment by dcbw** [comment]
See https://issues.redhat.com/browse/DPTP-3458 to figure out why our Dockerfile rename changes keep getting overwritten.

**Comment by martinkennelly** [comment]
depends on https://github.com/openshift/release/pull/39364

**Comment by dcbw** [comment]
openshift/release PR merged so

/retest

**Pull request closed without merging**